ci/fetch-artifacts.sh

```diff
@@ -2,48 +2,92 @@
+# The curent pwd should be build/scripts when running this script.
 
-WINDOWS_ARTIFACT_URL="https://ci.appveyor.com/api/buildjobs/by4cpqmul79tp9ag/artifacts/build/scripts/htm_core-6866bd13ece35608313709e82c29b2260ee7be73-windows64.zip" #FIXME make the URLs automated, needs API?
-OSX_ARTIFACT_URL="https://3238-118031881-gh.circle-artifacts.com/0/tmp/uploads/htm_core-7b7b53c798b3795088ce772e96c6994068fc1302-darwin64.tar.gz"
+
+export VERSION=`cat ../../VERSION`
+echo "Released version: $VERSION"
+echo
+
+export TRAVIS_FILE="htm_core-$VERSION-linux64.tar.gz"
+
+#WINDOWS_ARTIFACT_URL="https://ci.appveyor.com/api/buildjobs/by4cpqmul79tp9ag/artifacts/build/scripts/htm_core-6866bd13ece35608313709e82c29b2260ee7be73-windows64.zip?branch=master" #FIXME make the URLs automated, needs API?
+#OSX_ARTIFACT_URL="https://3238-118031881-gh.circle-artifacts.com/0/tmp/uploads/$CIRCLE_FILE"
+
+
+echo
+echo "Fetching Windows artifacts:"
+export APPVEYOR_TOKEN="v2.5tido3qf80n33le5bi3e"  # token by David Keeney.  TODO: replace with project token.  Needs Admin permissions.
+export APPVEYOR_FILE="htm_core-$VERSION-windows64.zip"
+export APPVEYOR_ARTIFACT_URL="https://ci.appveyor.com/api/projects/htm-community/htm.core/artifacts/build/scripts/$APPVEYOR_FILE?branch=master" 
 
 echo "Fetching Windows artifacts:" 
-retry=false
+# Perform authentication
+
 for i in `seq 10`; do
-  echo "Downloading artifacts. Try $i"
-  wget "$WINDOWS_ARTIFACT_URL" || retry=true
-  if [[ x"$retry" == "xtrue" ]]; then
-    echo "sleep for 5min is good"
-    sleep 300 # wait 5min to retry. Other CI needs to finish its build
-    retry=false
+  echo "  Downloading artifacts. Try number $i"
+  # get the headers from the query for the latest master build.  
+  # This is most likely a redirect to the real URL of the artifact.
+  # So, Locate the line for "location:" and use the word in quotes as the real URL.
+  # Then we can use the -O option for the filename of the file it found.
+  page=`curl -v -X GET -I -H "Authorization: Bearer $APPVEYOR_TOKEN" $APPVEYOR_ARTIFACT_URL 2>/dev/null`
+echo "  $page \n"
+  newurl=`echo $page | grep Location: | sed -e 's/^.*: "//' -e 's/".*//'`
+  if [[ ! -z "$newurl" ]]; then
+    # We have one, fetch the file
+    curl -O $newurl
+    ex=$?
+    if test "$ex" == "0"; then  
+      echo "  happy"
+      ls *.zip
+      break
+    fi
   else
-    echo "happy"
-    ls *.zip
-    break
+    echo "URL empty."
   fi
+  echo "  sleep for 5min is good"
+  sleep 300 # wait 5min to retry. Other CI needs to finish its build
 done
 
-echo "Fetching OSX artifacts:"
-retry=false
+export CIRCLE_FILE="htm_core-$VERSION-darwin64.tar.gz"
+export CIRCLE_TOKEN='?circle-token=e4ec350f46372b0f9e0dc0f35453ee8c55d98838'    #token by david keeney  TODO, replace with project token.   Needs Admin permissions.
+export CIRCLE_URL="https://circleci.com/api/v1.1/project/github/htm-community/htm.core/latest/artifacts?circle-token=$CIRCLE_TOKEN&limit=1&branch=master&filter=successful"
```

Comment:
TODO I should do that token for CircleCI too

Reply:
CircleCI tokens from https://circleci.com/gh/htm-community/htm.core/edit#api 

I have an API token for Circle, researching how safe is it to post it publicly, I cannot find equivalent of `secret: ` as in AppV and Travis..